### PR TITLE
Adapt to new start-up relocation strategy

### PR DESCRIPTION
**Pull request by Yoga07**

### CODE REVIEW

File: src/node/node_duties/network_events.rs [hunk 1]
```diff
@@ -54,29 +58,25 @@ impl NetworkEvents {
...
-                if let ConnectionEvent::First = event {
-                    self.duty_cfg.setup_as_first().await
+                if startup_relocation {
+                    trace!("New node has joined the network");
+                    Some(ProcessNewMember(XorName(name.0)).into())
+                } else if let Some(prev_name) = previous_name {
+                    Some(
+                        ProcessRelocatedMember {
+                            old_node_id: XorName(name.0),
+                            new_node_id: XorName(prev_name.0),
```
Comment (oetyng) [inline src/node/node_duties/network_events.rs:75]: Seems `name` and `prev_name` got switched here?

File: Cargo.toml [hunk 2]
```diff
@@ -31,7 +31,7 @@ pickledb = "~0.4.0"
 quick-error = "1.2.2"
 rand = "~0.7.3"
 rand_chacha = "~0.2.2"
-sn_routing = { git = "https://github.com/maidsafe/sn_routing" }
+sn_routing = { git = "https://github.com/yoga07/sn_routing", branch = "start-up" }
```
Comment (joshuef) [inline Cargo.toml:34]: can we not keep this pointing to master? Is routing missing something?

**Review by joshuef — APPROVED** [review]

**Review by joshuef — CHANGES_REQUESTED** [review]
clippy :|

File: src/node/node_duties/network_events.rs (hunk 1 above)
Comment (Yoga07) [reply src/node/node_duties/network_events.rs:75]: Ah Nice catch Ed! Will tend to it.

**Comment by joshuef** [comment]
incorporated into https://github.com/maidsafe/sn_node/pull/1135

**Pull request closed without merging**